### PR TITLE
feat(ShidokaIcon): replaced carbon icon to shidoka icon

### PR DESCRIPTION
**Pull request by kumarivijeta**
## Summary

- Removed Icon component
- Carbon Icon replaced with Shidoka icon
- Added migration docs
- Impacted Component List
  - Accordian
  -Button with Icon
  -Link with Icon
  -Split Button & with Icon
- Added Icon mdx under Foundation category

## ADO Story

[TASK 1972305](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1972305)

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [ ] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.

## Screenshot

image

image

image

image

image

image

### CODE REVIEW

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
1. The carbon-icons dependencies still need to be removed from package.json.
2. Remember to include the `BREAKING CHANGE:` footer in your commit messages next time.

**Review by srpriyankashetty — APPROVED** [review]
Verified the Accordion icons, numbering, Expand and collapse and looks fine
Verified the Button icons in Button with icon, icon only, default button in the bottom, scroll button and with secondary button
Verified the Link text icons
Verified the Split button icons and looks fine
Verified the size of the button and arrow/icon size changes as per the selection
Verified that disabling, disables the whole button with arrow/icon
Verified that disabled buttons arrow does not change on expand or collapse

**Review by kumarivijeta — COMMENTED** [review]
@brian-patrick-3  
**FYI**
The red marked icons in screenshot are not available in shidoka icons. I have requested Diana to get it added in shidoka icon library.

image

**Comment by brian-patrick-3** [comment]
Those icons are just meant to be examples, so I don't think they need to map 1:1. Maybe just use circle stroke and warning filled.

File: src/components/icon/Icon.mdx [hunk 1]
```diff
@@ -1,122 +0,0 @@
...
-
-```js
-import iconName from '@kyndryl-design-system/shidoka-foundation/assets/svg/<filename>';
-```
-
-The method for placing them in your app at this point will vary based on the framework/loader.
-
-<br />
-
-### Gallery
```
Comment (brian-patrick-3) [inline src/components/icon/Icon.mdx:77]: One small problem here, we still need to document the usage of the logo SVG assets somewhere. Logout and switcher can be removed.

**Pull request merged**